chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,233 @@
+import numpy
+
+import cupy
+
+from chainer import cuda
+from chainer import reporter
+from chainer import training
+from chainer.training import extension
+
+
+def _iterable(x):
+    if isinstance(x, (list, tuple)):
+        return x
+    return x,
+
+
+def _prefix_statistics(prefix, stats):
+    """Prefix all keys in a statistic dictionary."""
+    for key in list(stats.keys()):
```

Comment:
It was required since we added new (key, value) pairs to the dict while iterating over it. A new dict is returned each time now instead.